provider/github/poster.go

```diff
@@ -192,3 +194,68 @@ func createReviewRequest(
 
 	return req, nil
 }
+
+// Status sets the Pull Request global status, visible from the GitHub UI
+func (p *Poster) Status(ctx context.Context, e lookout.Event, status lookout.AnalysisStatus) error {
+	switch ev := e.(type) {
+	case *lookout.ReviewEvent:
+		if ev.Provider != Provider {
+			return ErrEventNotSupported.Wrap(
+				fmt.Errorf("unsupported provider: %s", ev.Provider))
+		}
+
+		return p.statusPR(ctx, ev, status)
+	default:
+		return ErrEventNotSupported.Wrap(fmt.Errorf("unsupported event type"))
+	}
+}
+
+// StatusCreator creates statuses on GitHub. *github.RepositoriesService
+// fulfills this interface.
+type StatusCreator interface {
+	// CreateStatus creates a new status for a repository at the specified
+	// reference. Ref can be a SHA, a branch name, or a tag name.
+	CreateStatus(ctx context.Context, owner, repo, ref string, status *github.RepoStatus) (
+		*github.RepoStatus, *github.Response, error)
+}
+
+var _ StatusCreator = &github.RepositoriesService{}
+
+func statusStrings(s lookout.AnalysisStatus) (string, string, error) {
+	switch s {
+	case lookout.ErrorAnalysisStatus:
+		return "error", "There was an error during the analysis", nil
+	case lookout.FailureAnalysisStatus:
+		return "failure", "The analysis result was negative", nil
+	case lookout.PendingAnalysisStatus:
+		return "pending", "The analysis is in progress", nil
+	case lookout.SuccessAnalysisStatus:
+		return "success", "The analysis was performed", nil
+	default:
+		return "", "", fmt.Errorf("unsupported AnalysisStatus %s", s)
+	}
+}
+
+func (p *Poster) statusPR(ctx context.Context, e *lookout.ReviewEvent, status lookout.AnalysisStatus) error {
+	owner, repo, _, err := p.validatePR(e)
+	if err != nil {
+		return err
+	}
+
+	statusStr, description, err := statusStrings(status)
+	if err != nil {
+		return err
+	}
+	targetURL := "https://github.com/src-d/lookout"
```

Comment:
hard code?

Reply:
For now I didn't see it clearly if we should put it in an env var, CLI argument, or field in the conf... so I left it like that as a placeholder.

I think this field is optional, it's the link for "Details" in the PR check list. If you think the hardcoded value looks too bad we can simply leave it unset, and add it later when we have a proper UI to link.

Reply:
yeah. I got it's a link. I'm okay to keep hard-coded value for now. But maybe we should move it to a constant on the top of the file so it will be easier to see/find later? I would do the same for `context` variable.

Reply:
Done in 3318d22.